cmd/puppet-db/root.go

```diff
@@ -138,24 +144,93 @@ func setCmdFlags(cmd *cobra.Command) {
 	cmd.PersistentFlags().StringP(
 		"cert",
 		"",
-		getDefaultCert(),
+		"",
 		"`Path` to client certificate for auth",
 	)
 	cmd.PersistentFlags().StringP(
 		"key",
 		"",
-		getDefaultKey(),
+		"",
 		"`Path` to client certificate for auth",
 	)
 
 	cmd.PersistentFlags().StringP(
 		"token",
 		"",
-		getDefaultToken(),
+		"",
 		"`Path` to RBAC token for auth (PE Only)",
 	)
 }
 
+func initConfig(cfgFile string) {
+	viper.SetConfigType("json")
+	err := readConfigFile(cfgFile)
+	if err != nil {
+		log.Error(err.Error())
+		os.Exit(1)
+	}
+}
+
+func readConfigFile(cfgFile string) error {
+	err := readGlobalConfigFile()
+	if err != nil {
+		return err
+	}
+
+	return mergeUserConfigFile(cfgFile)
+}
+
+func getGlobalConfigFile() (string, error) {
+	puppetLabsDir, err := PuppetLabsDir()
+	if err != nil {
+		return puppetLabsDir, err
+	}
+
+	globalConfigFile := filepath.Join(puppetLabsDir, "client-tools", "puppet-code.conf")
+	return globalConfigFile, nil
+}
+
+func readGlobalConfigFile() error {
+	globalConfigFile, err := getGlobalConfigFile()
+	if err != nil {
+		return err
+	}
+
+	_, err = os.Stat(globalConfigFile)
+	if err != nil {
+		log.Debug(fmt.Sprintf("Failed reading global config file: %s", err.Error()))
+		return nil
+	}
+	viper.SetConfigFile(globalConfigFile)
+	return viper.ReadInConfig()
+}
+
+func getDefaultConfig() string {
+	usr, err := user.Current()
+	if err != nil {
+		// TODO can this really happen? what should we return here
+		log.Error(err.Error())
+		return ""
+	}
```

Comment:
@ciprianbadescu any input on this part?

Reply:
I think should be like this for consistency (err from user.Current should be handled)